src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -281,7 +317,31 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
                             ConsoleWriteLine("\n\n");
 
                             FileInfo fileInfo = new(output.FullName);
-                            Task copyTask = session.EventStream.CopyToAsync(fs);
+                            EventMonitor eventMonitor = null;
+                            Task copyTask = null;
+                            if (hasStoppingEventProviderName)
+                            {
+                                eventMonitor = new(
+                                    stoppingEventProviderName,
+                                    stoppingEventEventName,
+                                    payloadFilter,
+                                    onEvent: (traceEvent) =>
+                                    {
+                                        shouldExit.Set();
+                                    },
+                                    onPayloadFilterMismatch: (traceEvent) =>
+                                    {
+                                        ConsoleWriteLine($"One or more field names specified in the payload filter for event '{traceEvent.ProviderName}/{traceEvent.EventName}' do not match any of the known field names: '{string.Join(' ', traceEvent.PayloadNames)}'. As a result the requested stopping event is unreachable; will continue to collect the trace for the remaining specified duration.");
```

Comment:
Does this clobber the tools output?

Reply:
No, `onPayloadFilterMismatch` should only be invoked once. This payload mismatch is only encountered if the payload filter the user specifies would never match the manifest, (e.g. user wants to match a trace event that has 5 payload values but the manifest dictates that the trace event will only ever have 4, or user passes in a payload name  that the trace event will never have as a payload name)

https://github.com/dotnet/diagnostics/blob/7dd062f0f3ab14d20b57d41a90d7997be063e20c/src/Microsoft.Diagnostics.Monitoring.EventPipe/EventMonitor.cs#L102-L103